docker/docker/src/main/java/org/arquillian/cube/docker/impl/docker/compose/ComposeBuilder.java

```diff
@@ -0,0 +1,57 @@
+package org.arquillian.cube.docker.impl.docker.compose;
+
+import static org.arquillian.cube.docker.impl.util.YamlUtil.asMap;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Set;
+
+import org.arquillian.cube.docker.impl.client.config.CubeContainer;
+import org.arquillian.cube.docker.impl.client.config.DockerCompositions;
+import org.arquillian.cube.docker.impl.client.config.Network;
+
+public class ComposeBuilder {
+
+	private DockerCompositions configuration;
+
+	private static final String SERVICES = "services";
+	private static final String NETWORKS = "networks";
+
+	private Path dockerComposeRootLocation;
+
+	ComposeBuilder(Path dockerComposeRootLocation) {
+		this(dockerComposeRootLocation, new DockerCompositions());
+	}
+
+	private ComposeBuilder(Path dockerComposeRootLocation,
+	                       DockerCompositions configuration) {
+		this.dockerComposeRootLocation = dockerComposeRootLocation;
+		this.configuration = configuration;
+	}
+
+	@SuppressWarnings("unchecked")
+	public DockerCompositions build(Map<String, Object> dockerComposeContainerDefinition) {
+		if (dockerComposeContainerDefinition.containsKey(SERVICES)) {
+			Map<String, Object> servicesDefinition = asMap(
+					dockerComposeContainerDefinition, SERVICES);
+			Set<String> keys = servicesDefinition.keySet();
+			for (String key : keys) {
+				ContainerBuilder services = new ContainerBuilder(this.dockerComposeRootLocation);
+				CubeContainer cubeContainer = services.build(asMap(servicesDefinition, key));
+				this.configuration.add(key, cubeContainer);
+			}
+		}
+		if (dockerComposeContainerDefinition.containsKey(NETWORKS)) {
```

Comment:
According to docker compose v2, if there is no network defined, one default is provided (https://docs.docker.com/compose/networking/) So what I would do is:

if no network provide, create one (it means creating a new network definition automatically) with the `current directory_default` and more important iterate over all cube definitions and provide a networkMode field with the name of this network.